exercises/java/src/main/java/com/adaptionsoft/games/uglytrivia/Game.java

```diff
@@ -1,169 +1,187 @@
-				System.out.println("Answer was correct!!!!");
-				purses[currentPlayer]++;
-				System.out.println(players.get(currentPlayer) 
-						+ " now has "
-						+ purses[currentPlayer]
-						+ " Gold Coins.");
-				
-				boolean winner = didPlayerWin();
-				currentPlayer++;
-				if (currentPlayer == players.size()) currentPlayer = 0;
-				
-				return winner;
-			} else {
-				currentPlayer++;
-				if (currentPlayer == players.size()) currentPlayer = 0;
-				return true;
-			}
-			
-			
-			
-		} else {
-		
-			System.out.println("Answer was corrent!!!!");
-			purses[currentPlayer]++;
-			System.out.println(players.get(currentPlayer) 
-					+ " now has "
-					+ purses[currentPlayer]
-					+ " Gold Coins.");
-			
-			boolean winner = didPlayerWin();
-			currentPlayer++;
-			if (currentPlayer == players.size()) currentPlayer = 0;
-			
-			return winner;
-		}
-	}
-	
-	public boolean wrongAnswer(){
-		System.out.println("Question was incorrectly answered");
-		System.out.println(players.get(currentPlayer)+ " was sent to the penalty box");
-		inPenaltyBox[currentPlayer] = true;
-		
-		currentPlayer++;
-		if (currentPlayer == players.size()) currentPlayer = 0;
-		return true;
-	}
-
-
-	private boolean didPlayerWin() {
-		return !(purses[currentPlayer] == 6);
-	}
-}
+package com.adaptionsoft.games.uglytrivia;
+
+import java.util.*;
+
+public class Game {
+    private static final String POP = "Pop";
+	private static final String SCIENCE = "Science";
+	private static final String SPORTS = "Sports";
+	private static final String ROCK = "Rock";
+
+    private LinkedList<String> popQuestions = new LinkedList<>();
+    private LinkedList<String> scienceQuestions = new LinkedList<>();
+    private LinkedList<String> sportsQuestions = new LinkedList<>();
+    private LinkedList<String> rockQuestions = new LinkedList<>();
+
+    private static final int END_SPACE = 11;
+    private static final int NUMBER_OF_SPACES = 12;
+    private static final int BOARD_STARTING_LOCATION = 0;
+    private static final int ZERO_COINS = 0;
+    private static final int TOTAL_NUMBER_OF_QUESTIONS = 50;
+
+    private Map<Integer, String> board = new HashMap<>();
+	private ArrayList<String> allPlayers = new ArrayList<>();
+
+    private int[] locationOnBoard = new int[6];
+    private int[] goldCoins = new int[6];
+    private boolean[] isInPenaltyBox = new boolean[6];
+
+    private int currentPlayer = 0;
+    private boolean isExitingPenaltyBox;
+
+    public  Game(){
+        addQuestions();
+        assignCategoriesToBoardSpaces();
+    }
+
+	public void addPlayer(String playerName) {
+	    allPlayers.add(playerName);
+	    locationOnBoard[numberOfPlayers()] = BOARD_STARTING_LOCATION;
+	    goldCoins[numberOfPlayers()] = ZERO_COINS;
+	    isInPenaltyBox[numberOfPlayers()] = false;
+
+	    System.out.println(playerName + " was added");
+	    System.out.println("They are player number " + numberOfPlayers());
+    }
+
+	public void rollDice(int currentRoll) {
```

Comment:
Nice drawing out of the separate functionality in this method to two separate methods.